wgpu/examples/wgpu-info/main.rs

```diff
@@ -0,0 +1,145 @@
+use std::{
+    process::{exit, Command},
+    time::Instant,
+};
+
+// Lets keep these on one line
+#[rustfmt::skip]
+fn print_info_from_adapter(adapter: &wgpu::Adapter, idx: usize) {
+    let info = adapter.get_info();
+    let downlevel = adapter.get_downlevel_properties();
+    let features = adapter.features();
+    let limits = adapter.limits();
+
+    println!("Adapter {}:", idx);
+    println!("\tBackend:   {:?}", info.backend);
+    println!("\tName:      {:?}", info.name);
+    println!("\tVendorID:  {:?}", info.vendor);
+    println!("\tDeviceID:  {:?}", info.device);
+    println!("\tType:      {:?}", info.device_type);
+    println!("\tCompliant: {:?}", downlevel.is_webgpu_compliant());
+    println!("\tFeatures:");
+    println!("\t\tDepth Clamping:                             {}", features.contains(wgpu::Features::DEPTH_CLAMPING));
```

Comment:
is it really not good to just debug-print the features? it would say something like `DEBUG_CLAMPING | TIMESTAMP_QUERY`.
I guess one thing that it gives you is a quick lookup for a question like "is something supported?"

Reply:
The string on high-supported cards gets _really_ long and wraps 3-4 times, so it's very hard to figure out which are aren't. This also makes it easier to see which _aren't_ supported, which is often more relevant.

Reply:
We can print this in a more automated way: we can iterate the bit, and for each bit print a separate line like you did here. Something like:
```rust
let count = wgt::Features::all().bits().trailing_ones();
for i in 0 .. count {
  let bit = wgt::Features::from_bits(1 << i);
  println!("\t\t{:?}: {}", bit, features.contains(bit));
}
```

Same for downlevel flags. Would save quite a bit of repetition here.
But yes, there is a trade-off. Your hand-written description is better.

Maybe we can have a hybrid approach?  We'd implement `Display` for the bitflags, putting your nice descriptions in there, but then just have a loop in here like I suggested, using that `Display` implementation?

Reply:
That seems reasonable, that also prevents this code from bitrotting as more features get added but no one remembers this exists. I think I may also explicitly destructure the limits structs so that adding new limits errors out here.